### PR TITLE
CombinedReducers MDRange support

### PR DESCRIPTION
**Pull request by dhollman**
Implement feature request in #3394

### CODE REVIEW

**Comment by dhollman** [comment]
@dalg24 I have no idea why clang-format changed some other things in the file that I didn't touch. Any thoughts?

**Review by dalg24 — COMMENTED** [review]
Why aren't you taking advantage of FTAD?

**Comment by dalg24** [comment]
> Why aren't you taking advantage of FTAD?

I pulled your changes and I see that my suggestion compiles but produces a bug.  Please educate me.

**Comment by crtrott** [comment]
OK I don't understand it either :-) @nliber @dhollman so if you could educate us what is going on maybe with a smaller example that would be awesome!

**Comment by masterleinad** [comment]
```
/Users/travis/build/kokkos/kokkos/core/unit_test/TestReduce.hpp:523:63: error: suggest braces around initialization of subobject [-Werror,-Wmissing-braces]
      Kokkos::MDRangePolicy<TEST_EXECSPACE, Kokkos::Rank<3>>({0, 0, 0},
                                                              ^~~~~~~
                                                              {      }
/Users/travis/build/kokkos/kokkos/core/unit_test/TestReduce.hpp:524:63: error: suggest braces around initialization of subobject [-Werror,-Wmissing-braces]
                                                             {nw, 1, 1}),
                                                              ^~~~~~~~
                                                              {       }
```

File: core/unit_test/TestReduce.hpp [hunk 1]
```diff
@@ -499,6 +508,27 @@ TEST(TEST_CATEGORY, int_combined_reduce) {
...
+  uint64_t nsum = (nw / 2) * (nw + 1);
+
+  int64_t result1 = 0;
+  int64_t result2 = 0;
+  int64_t result3 = 0;
+
+  Kokkos::parallel_reduce(
+      "int_combined_reduce_mdrange",
+      Kokkos::MDRangePolicy<TEST_EXECSPACE, Kokkos::Rank<3>>({0, 0, 0},
+                                                             {nw, 1, 1}),
```
Comment (masterleinad) [inline core/unit_test/TestReduce.hpp:524]:
```suggestion
      Kokkos::MDRangePolicy<TEST_EXECSPACE, Kokkos::Rank<3>>({{0, 0, 0}},
                                                             {{nw, 1, 1}}),
```

**Comment by dhollman** [comment]
@dalg24 @crtrott There's actually nothing really complicated going on here. You can't use FTAD here because you can't deduce a variadic parameter pack that's not at the end of the parameter list, so you have to give the first parameter pack explicitly at the call site.

**Comment by crtrott** [comment]
Ah ok I get it now.

**Review by crtrott — APPROVED** [review]

**Pull request merged**